### PR TITLE
Fix Ktor 937

### PR DESCRIPTION
**Pull request by hfhbd**
# Subsystem #
CIO JVM TLSHandshake Clientrequest

Fixing: [KTOR-937](https://youtrack.jetbrains.com/issue/KTOR-937)

# Motivation #
Currently, CIO crashes when a server requests a client certificate and restricts the authority to some specific. CIO reads the size bytes for all the authorities, but did not add the size of each authorities to the position counter. As a result, CIO tries to read more bytes than available and crashes.

# Solution #
Add missing 2 Bytes to position after reading the size of authority

### CODE REVIEW

File: ktor-network/ktor-network-tls/jvm/src/io/ktor/network/tls/TLSClientHandshake.kt [hunk 1]
```diff
@@ -488,3 +463,35 @@ private fun generateECKeys(curve: NamedCurve, serverPoint: ECPoint): EncryptionI
 
     return EncryptionInfo(serverPublic, clientKeys.public, clientKeys.private)
 }
+
+internal fun clientCertificateRequest(packet: ByteReadPacket): CertificateInfo {
```
Comment (andrey-vasilyev) [inline ktor-network/ktor-network-tls/jvm/src/io/ktor/network/tls/TLSClientHandshake.kt:467]:
```suggestion
internal fun readClientCertificateRequest(packet: ByteReadPacket): CertificateInfo {
```

File: ktor-network/ktor-network-tls/jvm/test/io/ktor/network/tls/tests/ClientCertificateRequestTest.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
...87 45 67 65 45 79 84 72 82 45 48 57 0 91 48 89 49 19 48 17 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 3 99 111 109 49 19 48 17 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 3 118 119 103 49 21 48 19 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 5 86 87 80 75 73 49 22 48 20 6 3 85 4 3 19 13 86 87 45 67 65 45 79 84 72 82 45 49 48 0 91 48 89 49 19 48 17 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 3 99 111 109 49 19 48 17 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 3 118 119 103 49 21 48 19 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 5 86 87 80 75 73 49 22 48 20 6 3 85 4 3 19 13 86 87 45 67 65 45 80 82 79 67 45 48 55 0 91 48 89 49 19 48 17 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 3 99 111 109 49 19 48 17 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 3 118 119 103 49 21 48 19 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 5 86 87 80 75 73 49 22 48 20 6 3 85 4 3 19 13 86 87 45 67 65 45 80 82 79 67 45 48 56 0 91 48 89 49 19 48 17 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 3 99 111 109 49 19 48 17 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 3 118 119 103 49 21 48 19 6 10 9 -110 38 -119 -109 -14 44 100 1 25 22 5 86 87 80 75 73 49 22 48 20 6 3 85 4 3 19 13 86 87 45 67 65 45 82 79 79 84 45 48 53"
```
Comment (andrey-vasilyev) [inline ktor-network/ktor-network-tls/jvm/test/io/ktor/network/tls/tests/ClientCertificateRequestTest.kt:14]: I guess `arrayOf<Byte>(3, 1, 2, 64, 0, 46,...).toByteArray()` will be a bit better

Comment (andrey-vasilyev) [reply ktor-network/ktor-network-tls/jvm/test/io/ktor/network/tls/tests/ClientCertificateRequestTest.kt:14]: But ideally we should figure out how to generate certificate request

Comment (hfhbd) [reply ktor-network/ktor-network-tls/jvm/test/io/ktor/network/tls/tests/ClientCertificateRequestTest.kt:14]: I simply copied a packet from a request... Of course I can refactor these mapping. 
But yes, I want to have the cert-request feature and tests in ktor too!

**Review by andrey-vasilyev — APPROVED** [review]

File: ktor-network/ktor-network-tls/jvm/src/io/ktor/network/tls/TLSClientHandshake.kt [hunk 3]
```diff
@@ -488,3 +463,35 @@ private fun generateECKeys(curve: NamedCurve, serverPoint: ECPoint): EncryptionI
 
     return EncryptionInfo(serverPublic, clientKeys.public, clientKeys.private)
 }
+
+internal fun readClientCertificateRequest(packet: ByteReadPacket): CertificateInfo {
```
Comment (andrey-vasilyev) [inline ktor-network/ktor-network-tls/jvm/src/io/ktor/network/tls/TLSClientHandshake.kt:467]:
```suggestion
   /**
    * RFC 5246, 7.4.4.  Certificate Request:
    *
    *     struct {
    *         ClientCertificateType certificate_types<1..2^8-1>;
    *         SignatureAndHashAlgorithm supported_signature_algorithms<2^16-1>;
    *         DistinguishedName certificate_authorities<0..2^16-1>;
    *     } CertificateRequest;
    */
internal fun readClientCertificateRequest(packet: ByteReadPacket): CertificateInfo {
```

**Review by e5l — APPROVED** [review]
Hey, @hfhbd. Great job, LGTM. It will be merged after the CI build.

**Review by andrey-vasilyev — APPROVED** [review]

**Pull request merged**